### PR TITLE
Fix lz4 library incompatibility in kafka-indexing-service extension

### PR DESCRIPTION
**Pull request by satishbhor**
Added support for Kafka 0.10.0.0 lz4 so that #3266  issue can be resolved.

### CODE REVIEW

File: extensions-core/kafka-indexing-service/pom.xml [hunk 1]
```diff
@@ -67,7 +67,7 @@
     <dependency>
       <groupId>org.apache.kafka</groupId>
       <artifactId>kafka_2.11</artifactId>
-      <version>0.9.0.0</version>
+      <version>0.10.0.0</version>
```
Comment (gianm) [inline extensions-core/kafka-indexing-service/pom.xml]: 0.10.2.0 is the latest, please use that if it works.

**Comment by gianm** [comment]
@satishbhor Btw, if you haven't already filled out a CLA, could you please do so at: http://druid.io/community/cla.html

File: extensions-core/kafka-indexing-service/pom.xml (hunk 1 above)
Comment (satishbhor) [inline extensions-core/kafka-indexing-service/pom.xml]: @gianm Thanks! I will  update Kafka indexing service document and will also check with Kafka 0.10.2.0

**Comment by satishbhor** [comment]
@gianm with minor changes in unit test cases it is working fine with Kafka 0.10.2.0 and also updated Kafka indexing service note in docs.

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java [hunk 2]
```diff
@@ -1008,7 +1008,7 @@ private void possiblyResetOffsetsOrWait(
         final TopicPartition topicPartition = outOfRangePartition.getKey();
         final long nextOffset = outOfRangePartition.getValue();
         // seek to the beginning to get the least available offset
-        consumer.seekToBeginning(topicPartition);
+        consumer.seekToBeginning(Lists.newArrayList(topicPartition));
```
Comment (leventov) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: Could use Collections.singletonList()?

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java [hunk 3]
```diff
@@ -1484,9 +1484,9 @@ private long getOffsetFromKafkaForPartition(int partition, boolean useEarliestOf
     }
 
     if (useEarliestOffset) {
-      consumer.seekToBeginning(topicPartition);
+      consumer.seekToBeginning(Lists.newArrayList(topicPartition));
```
Comment (leventov) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java]: Same

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java (hunk 2 above)
Comment (satishbhor) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: This file also contains Lists.newArrayList() in other places as well, so shall I replace those as well ?

Comment (leventov) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: The other place uses Iterable as argument, it's a different case

Comment (satishbhor) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: I mean to say statements like, 

`consumer.assign(Lists.newArrayList(topicPartition));`

Comment (leventov) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: Yes

Comment (satishbhor) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: Thanks!

Comment (satishbhor) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: Also in  #4135 you mentioned prefer `Collections.emptyList()`  over `Arrays.asList()`  then shall i also replace `Lists.newArrayList()` with `Collections.emptyList()` ?

Comment (leventov) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: It wasn't me, but yes, please

Comment (satishbhor) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: Thanks!

Comment (satishbhor) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java]: @leventov my apologize `Collections.emptyList()` can't be used as lists are being manipulated in code.

**Comment by satishbhor** [comment]
Thanks @gianm for accepting changes.

**Pull request merged**